src/Tickets/Order_Modifiers/Checkout/Abstract_Fees.php

```diff
@@ -0,0 +1,339 @@
+		$ticket_ids = array_map(
+			function ( $item ) {
+				return $item['ticket_id'];
+			},
+			$items
+		);
+
+		// Fetch related ticket fees and automatic fees.
+		$related_ticket_fees = $this->order_modifiers_repository->find_relationship_by_post_ids( $ticket_ids, $this->modifier_type );
+		$automatic_fees      = $this->order_modifiers_repository->get_all_automatic_fees();
+
+		// Combine the fees and remove duplicates.
+		$combined_fees = $this->extract_and_combine_fees( $related_ticket_fees, $automatic_fees );
+
+		// Cache the combined fees for future use.
+		wp_cache_set( $cache_key, $combined_fees, 'combined_fees', 120 );
+
+		return $combined_fees;
+	}
+
+	/**
+	 * Extracts and combines fees from related ticket fees and automatic fees, removing duplicates.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $related_ticket_fees The related ticket fees.
+	 * @param array $automatic_fees The automatic fees.
+	 *
+	 * @return array The combined array of fees.
+	 */
+	protected function extract_and_combine_fees( array $related_ticket_fees, array $automatic_fees ): array {
+		$all_fees = array_merge( $related_ticket_fees, $automatic_fees );
+
+		$unique_fees = [];
+		foreach ( $all_fees as $fee ) {
+			$id = $fee->id;
+
+			if ( ! isset( $unique_fees[ $id ] ) ) {
+				$unique_fees[ $id ] = [
+					'id'           => $id,
+					'raw_amount'   => $fee->raw_amount ?? 0,
+					'display_name' => $fee->display_name,
+					'sub_type'     => $fee->sub_type,
+				];
+
+				// Use the stored subtotal in the fee calculation.
+				$unique_fees[ $id ]['fee_amount'] = $this->manager->apply_fees_to_item( $this->subtotal, $unique_fees[ $id ] );
+			}
+		}
+
+		return array_values( $unique_fees );
+	}
+
+	/**
+	 * Adds fees as separate items in the cart.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $items    The current items in the cart.
+	 * @param Value $subtotal The calculated subtotal of the cart items.
+	 *
+	 * @return array Updated list of items with fees added.
+	 */
+	protected function append_fees_to_cart( array $items, Value $subtotal ) {
+		if ( self::$fees_appended ) {
+			return $items;
+		}
+
+		if ( empty( $items ) ) {
+			return $items;
+		}
+
+		$this->subtotal = $subtotal;
+
+		// Get all the combined fees for the items in the cart.
+		$fees = $this->get_combined_fees_for_items( $items );
+
+		// Track the fee_ids that have already been added to prevent duplication.
+		$existing_fee_ids = [];
+
+		foreach ( $items as $item ) {
+			if ( isset( $item['fee_id'] ) ) {
+				$existing_fee_ids[ $item['fee_id'] ] = 1;
+			}
+		}
+
+		// Loop through each fee and append it to the $items array if it's not already added.
+		foreach ( $fees as $fee ) {
+			// Skip if this fee has already been added to the cart.
+			if ( array_key_exists( $fee['id'], $existing_fee_ids ) ) {
+				continue;
+			}
+
+			// Append the fee to the cart.
+			// @todo - Review what needs to be sent. Some of these are used so wp_pluck doesn't cause a warning.
```

Comment:
I wanted to point this comment out. 

I know the biggest issue we had originally was the "Orders" page in the dashboard. I think the data below is good enough. However, will `ticket_id` and `event_id` 0 cause any issues in the future?

From the testing we did originally, we shouldn't have a problem anywhere else. If we do have a problem we need to incorporate the `type` check.